packages/lesswrong/components/notifications/NotificationTypeSettings.tsx

```diff
@@ -36,10 +43,12 @@ const NotificationTypeSettings = ({ path, value, label, classes }, context) => {
         onChange={(event) =>
           modifyValue({ channel: event.target.value })}
       >
-        {!notificationType.mustBeEnabled && <MenuItem value="none">Don't notify</MenuItem>}
+        {notificationType.allowedChannels?.length ?
+          notificationType.allowedChannels.map(channel => channelOptions[channel]) : <></>}
```

Comment:
I've replaced the `mustBeEnabled` option with `allowedChannels` (where `mustBeEnabled: true` is the same as `allowedChannels: ["onsite", "email", "both"]`